src/main/java/duke/commands/DukeCommandList.java

```diff
@@ -0,0 +1,74 @@
+package duke.commands;
+
+import duke.exceptions.DukeExceptionIllegalArgument;
+import duke.storage.FileLoader;
+import duke.tasks.Event;
+import duke.tasks.Todo;
+import duke.tasks.Deadline;
+import duke.tasks.Task;
+import duke.tasks.TaskList;
+import duke.ui.Ui;
+
+import java.util.ArrayList;
+
+/**
+ * List command.
+ *
+ * If 'toSort' is provided during construction, the list view will be
+ * sorted in order of Todo < Event (chronological) < Deadline (chronological).
+ */
+public class DukeCommandList extends DukeCommand {
+
+    protected boolean toSort;
+
+    public DukeCommandList(String arg) {
+        toSort = arg.equals("sort");
+    }
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, FileLoader loader) throws DukeExceptionIllegalArgument {
+        if (tasks.size() == 0) {
+            ui.showMessage("There are no tasks in your list.");
+        } else if (toSort) {
+            ArrayList<Task> view = new ArrayList<>();
+            for (int i = 0; i < tasks.size(); i++) {
+                view.add(tasks.getTask(i));
+            }
+
+            // Sort in order: Todo < Event (chronological) < Deadline (chronological)
+            view.sort((t1, t2) -> {
```

Comment:
Store the lambda inside a function?

Reply:
Perhaps, but this works pretty well at the moment